src/mjolnir/pbfgraphparser.cc

```diff
@@ -78,17 +78,30 @@ struct graph_callback : public OSMPBF::Callback {
 
   virtual void
   node_callback(uint64_t osmid, double lng, double lat, const OSMPBF::Tags& tags) override {
-    // Check if it is in the list of nodes used by ways
-    if (!shape_.get(osmid)) {
-      return;
-    }
-
     // Get tags
     Tags results = lua_.Transform(OSMType::kNode, tags);
     if (results.size() == 0) {
       return;
     }
 
+    if (bss_nodes_) {
```

Comment:
this is indeed a much better way to do this work. sorry for not catchign it earlier but thank you so much for fixing it!